src/fidesops/ops/api/v1/endpoints/saas_config_endpoints.py

```diff
@@ -294,47 +292,28 @@ def instantiate_connection_from_template(
             detail=f"SaaS connector type '{saas_connector_type}' is not yet available in Fidesops. For a list of available SaaS connectors, refer to {CONNECTION_TYPES}.",
         )
 
-    if DatasetConfig.filter(
-        db=db,
-        conditions=(DatasetConfig.fides_key == template_values.instance_key),
-    ).count():
-        raise HTTPException(
-            status_code=HTTP_400_BAD_REQUEST,
-            detail=f"SaaS connector instance key '{template_values.instance_key}' already exists.",
-        )
-
     try:
-        connection_config: ConnectionConfig = (
-            create_connection_config_from_template_no_save(
-                db, connector_template, template_values
-            )
+        connection_config, dataset_config = add_saas_connector(
+            db, connector_template, template_values, saas_connector_type
         )
     except KeyOrNameAlreadyExists as exc:
         raise HTTPException(
             status_code=HTTP_400_BAD_REQUEST,
             detail=exc.args[0],
         )
-
-    connection_config.secrets = validate_secrets(
-        template_values.secrets, connection_config
-    ).dict()
-    connection_config.save(db=db)  # Not persisted to db until secrets are validated
-
-    try:
-        dataset_config: DatasetConfig = create_dataset_config_from_template(
-            db, connection_config, connector_template, template_values
-        )
+    except HTTPException as exc:
```

Comment:
propagate up an HTTP exception that's thrown from the underlying function call, since that's ready to go back to the client, no need to wrap it further